Updated OpenCL test package.

caseymcc opened the pull request:
Removed support for Android as it should be linked from the NDK.
Removed support for Apple as OpenCL support stops at 1.2
Removed VC 9-10 as the headers contain some code that they don't understand.
Removed MinGW on windows as it links to the native direct x which has some VC specific code that gcc does not understand.

caseymcc commented:
Something wrong in the merge looking into it.

caseymcc commented:
Looks like test are passing, assuming everything else is good, this should be good to go.

ruslo commented on .travis.yml:
```diff
@@ -42,8 +42,6 @@ matrix:
     - os: linux
       env: PROJECT_DIR=examples/OpenCL TOOLCHAIN=gcc-4-8
     - os: linux
-      env: PROJECT_DIR=examples/OpenCL TOOLCHAIN=android-ndk-r10e-api-19-armeabi-v7a-neon
```
Please leave some info about build. At least link to the broken job.

caseymcc replied:
Not sure what you mean? Do you want me to comment out the tool chain and add a remark as to why it is not run?

ruslo replied:
Here is the link to the broken job:
* https://travis-ci.org/ingenue/hunter/jobs/180306953

Add it as a comment so if somebody want to take a look at the problem.

ruslo commented on cmake/configs/default.cmake:
```diff
@@ -66,6 +66,7 @@ hunter_config(Libssh2 VERSION 1.7.0)
 hunter_config(Lua VERSION 5.3.2)
 hunter_config(MySQL-client VERSION 6.1.6)
 hunter_config(OpenBLAS VERSION 0.2.19-p0)
+hunter_config(OpenCL VERSION 2.1-p0)
```
It doesn't belong here, see [README](https://github.com/ingenue/hunter). Just wrote some info: https://github.com/ingenue/hunter/pull/42#discussion_r90889722

caseymcc replied:
I was a bit confused about how the test were being run without this, guess a little reading pays off. I am assuming example code needs to be removed as well?

caseymcc commented:
I think I have made the changes you wanted.

caseymcc committed:
Updated OpenCL test package.
Removed support for Android as it should be linked from the NDK.
Removed support for Apple as OpenCL support stops at 1.2
Removed VC 9-11 as the headers contain some code that they don't understand.
Removed MinGW on windows as it links to the native direct x which has some VC specific code that gcc does not understand.

caseymcc commented:
Added links to failed toolchains.
